builder/builder.go

```diff
@@ -199,15 +204,143 @@ func (b *Builder) startRegistryProxy(ctx context.Context, caps apicaps.CapSet) (
 		doneCh <- struct{}{}
 	}()
 
+	isDarwin := runtime.GOOS == "darwin"
+
+	if isDarwin {
+		darwinProxyPort, err := b.startDarwinProxyHelper(ctx, registryPort)
+		if err != nil {
+			b.opt.Console.Warnf("Failed to start registry proxy support container: %v", err)
+			err := b.stopDarwinProxyHelper(ctx) // Clean up dangling container on failure.
+			if err != nil {
+				b.opt.Console.Warnf("Failed to stop registry proxy support container: %v", err)
+			}
+			return nil, false
+		}
+		b.opt.Console.VerbosePrintf("Started support container for registry proxy on port %d", darwinProxyPort)
+		addr = fmt.Sprintf("127.0.0.1:%d", darwinProxyPort)
+	}
+
 	b.opt.LocalRegistryAddr = addr
 
-	return func() {
+	closeFn := func() {
```

Comment:
Just a question: is this function passed along to the cleanup collection somewhere? It might be useful to do, to help shut things down cleanly in case of cancellation.

Reply:
At the moment it's called when the [build function exits](https://github.com/earthly/earthly/blob/mh%2Fregistry-proxy-mac/builder/builder.go#L372).

Reply:
Ok, then that is probably good enough!